src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -146,7 +146,7 @@ private Uri CreateAuthorizationUri(bool addVerifier = false, bool addState = fal
 
             if (addState)
             {
-                _state = Guid.NewGuid().ToString();
+                _state = Guid.NewGuid().ToString() + Guid.NewGuid().ToString();
```

Comment:
Consider adding a comment here on why this is done, as it is not obvious from reading the code.